src/web/services/StudentProfile.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { StudentDetails } from '../app/pages-instructor/student-profile/student-profile';
+import { MessageOutput } from '../types/api-output';
+import { StudentProfileUpdateRequest } from '../types/api-request';
+import { HttpRequestService } from './http-request.service';
+
+/**
+ * Handles student profile related logic provision.
+ */
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StudentProfileService {
```

Comment:
You can use `ng g s StudentProfile` so that Angular CLI will generate corresponding test file also. Instead of copy here.